decidim-budgets/db/migrate/20200714103519_move_budgets_to_own_model.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class MoveBudgetsToOwnModel < ActiveRecord::Migration[5.2]
+  class Component < ApplicationRecord
+    self.table_name = :decidim_components
+  end
+
+  class Budget < ApplicationRecord
+    self.table_name = :decidim_budgets_budgets
+  end
+
+  class Project < ApplicationRecord
+    self.table_name = :decidim_budgets_projects
+  end
+
+  class Order < ApplicationRecord
+    self.table_name = :decidim_budgets_orders
+  end
+
+  def up
+    Project.reset_column_information
+    Order.reset_column_information
+
+    budget_components.each do |component|
+      resource = create_budget_resource_from(component)
+
+      if resource
```

Comment:
there is any possibility of getting an empty `resource` here?

Reply:
I don't think so, if a resource fails on `create!` an exception will be thrown. I'll remove the `if resource`